plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts

```diff
@@ -357,26 +365,36 @@ export class HapiInstrumentation extends BasePlugin<typeof Hapi> {
     const instrumentation: HapiInstrumentation = this;
     if (route[handlerPatched] === true) return route;
     route[handlerPatched] = true;
-    if (typeof route.handler === 'function') {
-      const handler = route.handler as Hapi.Lifecycle.Method;
+    let handler;
+    if (route.handler) {
+      handler = route.handler;
+    } else if (route.options && route.options.handler) {
+      handler = route.options.handler;
+    }
+    if (typeof handler === 'function') {
+      const oldHandler = handler as Hapi.Lifecycle.Method;
       const newHandler: Hapi.Lifecycle.Method = async function (
         request: Hapi.Request,
         h: Hapi.ResponseToolkit,
         err?: Error
       ) {
         if (instrumentation._tracer.getCurrentSpan() === undefined) {
-          return await handler(request, h, err);
+          return await oldHandler(request, h, err);
         }
         const metadata = getRouteMetadata(route, pluginName);
         const span = instrumentation._tracer.startSpan(metadata.name, {
           attributes: metadata.attributes,
         });
-        const res = await handler(request, h, err);
+        const res = await oldHandler(request, h, err);
         span.end();
 
         return res;
       };
-      route.handler = newHandler;
+      if (route.handler) {
+        route.handler = newHandler;
+      } else if (route.options && route.options.handler) {
```

Comment:
@obecny Just wondering if that will affect the backwards compatibility on previous node version? Since the `?` is a new feature in ES2020 which is only fully supported in node 14

Reply:
you are writing in typescript which is then compiled

Reply:
ahhh ok right.